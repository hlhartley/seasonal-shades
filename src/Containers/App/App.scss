body {
    display: grid;
}

.color-search {
    background-color: #f1d8c8;
    padding-top: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: 20px;
}

.submit-btn {
    background-color: #dcb6aa;
    height: 27px;
    margin-left: 7px;
    border-radius: 5px;
}

.fa-search {
    margin-right: 10px;
    font-size: 20px;
}

.search-input {
    width: 200px;
    padding: 5px 5px 5px 10px;
}

