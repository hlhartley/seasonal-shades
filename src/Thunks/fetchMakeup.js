import { getLipstick, getLipstickColors, setLoading, setError } from '../Actions/index';
import { API } from '../Helpers/requests';

export const fetchMakeup = (path) => {
    return async (dispatch) => {
        try {
            dispatch(setLoading(true));
            const colors = {};
            const items = {};
            const result = await API(path);
            result.forEach((item) => {
                item.product_colors.forEach(color => {
                    // colors[color.hex_value] = item.id;
                    if (!colors[color.hex_value]) {
                        colors[color.hex_value] = [];
                    } 
                    colors[color.hex_value].push(item.id);
                })
                items[item.id] = item;
            });
            switch(path) {
                case 'lipstick':
                    dispatch(getLipstick(items));
                    dispatch(getLipstickColors(colors));
            }
        } catch (error) {
            dispatch(setError(error));
        }
        dispatch(setLoading(false));
    }
}
