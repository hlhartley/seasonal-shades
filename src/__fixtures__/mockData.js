export const allColors = {
    522: {
        product: {},
        hexcode: '#EED8BE'
    },
    771: {
        product: {},
        hexcode: '#BD9E9B' 
    },
        'casablanca': {
        product: {},
        hexcode: '#444446'
    }
};

export const mockState = {
    allColors: {
        522: {
            product: {},
            hexcode: '#EED8BE'
        },
        771: {
            product: {},
            hexcode: '#BD9E9B' 
        },
        'casablanca': {
            product: {},
            hexcode: '#444446'
        }
    },
    favorites: {
        '001lune': true,
        '001petal': true,
    },
    blushColors: undefined,
    eyeshadowColors: undefined,
    lipstickColors: [],
    nail_polishColors: undefined,
    moreColors: undefined,
};

export const mockFavorites = {
    '001lune': true,
    '001petal': true, 
}