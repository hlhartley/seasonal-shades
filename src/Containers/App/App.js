import React, { Component } from 'react';
import '../../../src/main.scss';
import Header from '../../Components/Header/Header';
import Navigation from '../Navigation/Navigation';
import Banner from '../../Components/Banner/Banner';
import ColorList from '../ColorList/ColorList';
import { setError } from '../../Actions';
import { connect } from 'react-redux';
import { Route, Switch, withRouter, Redirect } from 'react-router-dom';
import Favorites from '../../Containers/Favorites/Favorites';
import ShowColor from '../ShowColor/ShowColor';
import { fetchMakeup } from '../../Thunks/fetchMakeup';
import PageNotFound from '../../Components/PageNotFound/PageNotFound';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      currentType: 'eyeshadow',
      inputValue: '',
    };
  };

  // componentDidMount() {
  //   this.props.fetchMakeup('eyeshadow')
  // }

  handleChange = (e) => {
    this.setState({ inputValue: e.target.value })
  }

  updateCurrentType = (type) => {
    this.setState({ currentType: type })
  }

  // hexCodeInput = (e) => {
  //   e.preventDefault();
  //   const hexCode = this.state.inputValue;
  //   const color = this.props.allColors[hexCode];
  //     console.log(color)
  // }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Header />
          <Navigation updateCurrentType={this.updateCurrentType}/>
          <Banner />
          <div className='hexcode-search'>
            <form onSubmit={this.hexCodeInput}>
                <input type="text" onChange={this.handleChange} placeholder='# + 6-digit hexcode'></input>
                <input type="submit" value="Submit" className='submit-btn'></input>
            </form>
          </div>
          <Switch>
            <Route exact path='/'>
              <Redirect to='/eyeshadow'/>
            </Route>
            <Route exact path='/favorites' render={()=> <Favorites />}/>
            <Route exact path={`/${this.state.currentType}`} render={()=> <ColorList type={this.state.currentType}/>}/>
            <Route path={`/${this.state.currentType}/:color`} render={({ match }) => {
                return <ShowColor color={match.params.color} type={this.state.currentType}/>
            }} />
            <Route path = '*' component = {PageNotFound} />
          </Switch>
        </header>
      </div>
    )
  };
};

export const mapStateToProps = (state) => ({
  nail_polishColors: state.nailpolishColors,
  lipstickColors: state.lipstickColors,
  blushColors: state.blushColors,
  eyeshadowColors: state.eyeshadowColors,
  allColors: state.allColors,
});

export const mapDispatchToProps = (dispatch) => ({
  fetchMakeup: (path) => dispatch(fetchMakeup(path))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
