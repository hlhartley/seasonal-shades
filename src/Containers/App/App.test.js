import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';
import { fetchMakeup } from '../../Thunks/fetchMakeup';
import { allColors, mockState } from '../../__fixtures__/mockData';

jest.mock('../../Thunks/fetchMakeup')

describe('App', () => { 
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App allColors={allColors} lipstickColors={[]} blushColors={undefined} eyeshadowColors={undefined} nail_polishColors={undefined} fetchMakeup={jest.fn()}/>)
  });

  describe('App Component', () => {
    it('should exist', () => {
      expect(wrapper).toBeDefined()
    });

    it('should match the correct snapshot with everything passed in correctly', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('should have a proper default state', () => {
      expect(wrapper.state()).toEqual({
        currentType: 'eyeshadow',
        colorInputValue: '',
      });
    });

    it('handleChange should setState with the correct input value', () => {
      const mockEvent = (value) => {
        return {
          target: {
            value,
          }
        }
      }
      const expected = {
        currentType: 'eyeshadow',
        colorInputValue: 'some input',
      }
      wrapper.instance().handleChange(mockEvent('some input'))
      expect(expected.colorInputValue).toEqual('some input')
    });

    it('updateCurrentType should setState with the correct type', () => {
      const expected = {
        currentType: 'blush',
        colorInputValue: '',
      }
      wrapper.instance().updateCurrentType('blush')
      expect(expected.currentType).toEqual('blush')
    });
  });

  describe('mapStateToProps', () => {
    it('should should return an object with all colors, blushColors, eyeshadowColors, lipstickColors, and nail_polishColors', () => {
    const expected = {
        allColors: {
            522: {
                product: {},
                hexcode: '#EED8BE'
            },
            771: {
                product: {},
                hexcode: '#BD9E9B' 
            },
            'casablanca': {
                product: {},
                hexcode: '#444446'
            }
        },
        blushColors: undefined,
        eyeshadowColors: undefined,
        lipstickColors: [],
        nail_polishColors: undefined,
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with fetchMakeup action when fetchMakeup is called', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchMakeup('eyeshadow')
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchMakeup('eyeshadow')
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  });
});