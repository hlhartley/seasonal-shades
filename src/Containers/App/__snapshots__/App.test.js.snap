// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the correct snapshot with everything passed in correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <Connect(Header) />
    <Connect(Navigation)
      updateCurrentType={[Function]}
    />
    <Banner
      type="eyeshadow"
    />
    <Connect(Search)
      handleChange={[Function]}
      type="eyeshadow"
    />
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          push={false}
          to="/eyeshadow"
        />
      </Route>
      <Route
        exact={true}
        path="/favorites"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/eyeshadow"
        render={[Function]}
      />
      <Route
        path="/eyeshadow/:color"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </header>
</div>
`;
